src/converse-notification.js

```diff
@@ -32,32 +32,34 @@ converse.plugins.add('converse-notification', {
             play_sounds: true,
             sounds_path: api.settings.get("assets_path")+'/sounds/',
             notification_icon: 'logo/conversejs-filled.svg',
-            notification_delay: 5000
+            notification_delay: 5000,
+            notify_nicknames_without_references: false
         });
 
-        _converse.shouldNotifyOfGroupMessage = function (message) {
+        _converse.shouldNotifyOfGroupMessage = function (message, data) {
             /* Is this a group message worthy of notification?
              */
-            let notify_all = api.settings.get('notify_all_room_messages');
-            const jid = message.getAttribute('from'),
-                resource = Strophe.getResourceFromJid(jid),
-                room_jid = Strophe.getBareJidFromJid(jid),
-                sender = resource && Strophe.unescapeNode(resource) || '';
-            if (sender === '' || message.querySelectorAll('delay').length > 0) {
-                return false;
-            }
+            const jid = message.getAttribute('from');
+            const room_jid = Strophe.getBareJidFromJid(jid);
+            const notify_all = api.settings.get('notify_all_room_messages');
             const room = _converse.chatboxes.get(room_jid);
-            const body = message.querySelector('body');
-            if (body === null) {
-                return false;
-            }
-            const mentioned = (new RegExp(`\\b${room.get('nick')}\\b`)).test(body.textContent);
-            notify_all = notify_all === true ||
-                (Array.isArray(notify_all) && notify_all.includes(room_jid));
-            if (sender === room.get('nick') || (!notify_all && !mentioned)) {
-                return false;
+            const resource = Strophe.getResourceFromJid(jid);
+            const sender = resource && Strophe.unescapeNode(resource) || '';
+            let is_mentioned = false;
+            const nick = room.get('nick');
+            
+            if (api.settings.get('notify_nicknames_without_references')) {
+                const body = message.querySelector('body');
+                is_mentioned = (new RegExp(`\\b${nick}\\b`)).test(body.textContent);
             }
-            return true;
+            
+            const is_referenced = data.attrs.references.map(r => r.value).includes(`@${nick}`);
```

Comment:
This line doesn't work because we don't include the `@` in the reference data.

Please add a test for the case where a message with references is received, to check that a user is properly notified.

Reply:
I added a test that is passing. The references array contains object with a key `value` which has the `@` sign, at least that is what the test environment showed.